airflow/models/dag.py

```diff
@@ -1613,7 +1613,8 @@ def _get_task_instances(
 
         # Do we want full objects, or just the primary columns?
         if as_pk_tuple:
-            tis = session.query(TI.dag_id, TI.task_id, TI.run_id, TI.map_index)
+            # try number is needed here since its used to construct TaskInstanceKey
+            tis = session.query(TI.dag_id, TI.task_id, TI.run_id, TI._try_number, TI.map_index)
```

Comment:
Is `try_number` really functionally needed, or just to fill the argument position? If it’s only the positional issue, we can instead do

```python
tis = session.query(TI.dag_id, TI.task_id, TI.run_id, TI.map_index)
# ... later ...
TaskInstanceKey(**cols._mapping) for cols in tis.all())
```

Relevant SQLAlchemy documentation: https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Row._mapping

Reply:
yes that's much neater. i'll do that instead. thanks!